front/src/utils/geometry.ts

```diff
@@ -38,3 +41,98 @@ export function lengthFromLineCoordinates(coordinates?: Position[]) {
   }
   return NaN;
 }
+
+/**
+ * Compute the distance between two Points,
+ * based on pythagore.
+ */
+export function distance(a: Position, b: Position): number {
+  return Math.sqrt((b[0] - a[0]) ** 2 + (b[1] - a[1]) ** 2);
+}
+
+export interface NearestPointOnLine extends Feature<Point> {
+  properties: {
+    index: number;
+    dist: number;
+    location: number;
+  };
+}
+/**
+ * This function has the same signature than the one of turf (except for multilines),
+ * but it is more precise.
+ *
+ * You can check those links for references :
+ * - https://github.com/Turfjs/turf/issues/1440
+ * - https://github.com/Turfjs/turf/issues/2023
+ *
+ * For each segment, we compute the height from the point of the triangle
+ * composed of the segment (BC) and the point (A),
+ * and we check if A' (projection of A on BC) is in the segment (BC) by computing the angle for B & C
+ *
+ * Math reminder on triangle A,B & C where AB is the lenght of the segment:
+ * - The height from the vextex A is `Ha = AB * SIN(AB,BC)`
+ * Angle of vertex B (cosines law) is `AngleB = arccos((AB² + BC² - AC²) / (2 * AB * BC))`
+ */
+export function nearestPointOnLine(
+  line: Feature<LineString> | LineString,
+  inPoint: Feature<Point> | Point | Position,
+  options?: { units?: Units }
+): NearestPointOnLine {
+  const pointA = getCoord(inPoint);
+  const lineCoords: Position[] =
+    line.type === 'Feature' ? line.geometry.coordinates : line.coordinates;
+
+  const closestPointPerSegment: Array<{ point: Position; dist: number; index: number }> = [];
+  for (let n = 1; n < lineCoords.length; n += 1) {
+    const pointB = lineCoords[n - 1];
+    const pointC = lineCoords[n];
+
+    const lengthAB = turfDistance(pointA, pointB);
+
+    if (lengthAB === 0) {
+      closestPointPerSegment.push({ point: pointB, index: n - 1, dist: lengthAB });
+    } else {
+      const lengthAC = turfDistance(pointA, pointC);
+      const lengthBC = turfDistance(pointB, pointC);
+      const angleB = Math.acos(
+        (lengthAB ** 2 + lengthBC ** 2 - lengthAC ** 2) / (2 * lengthAB * lengthBC)
+      );
+      const angleC = Math.acos(
+        (lengthAC ** 2 + lengthBC ** 2 - lengthAB ** 2) / (2 * lengthAC * lengthBC)
+      );
+
+      if (angleB >= 90 * (Math.PI / 180)) {
+        closestPointPerSegment.push({ point: pointB, index: n - 1, dist: lengthAB });
+      } else if (angleC >= 90 * (Math.PI / 180)) {
+        closestPointPerSegment.push({ point: pointC, index: n - 1, dist: lengthAC });
+      } else {
+        const heightA = lengthAB * Math.sin(angleB);
+
+        // A' is the projection of A on the BC line
+        // we compute the BA' distance thanks to pythagore
+        const lengthBAPrime = Math.sqrt(lengthAB ** 2 - heightA ** 2);
```

Comment:
Here, you can directly compute BA' using sin(b)
```suggestion
        // A' is the projection of A on the BC line
        const lengthBAPrime = lengthAB * Math.cos(angleB);
```